modules/postgresql/main.tf

```diff
@@ -41,9 +41,15 @@ resource "google_sql_database_instance" "default" {
     disk_size       = "${var.disk_size}"
     disk_type       = "${var.disk_type}"
     pricing_plan    = "${var.pricing_plan}"
-    user_labels     = "${var.user_labels}"
     database_flags  = ["${var.database_flags}"]
 
+    // Define a label to force a dependency to the creation of the network peering.
+    // Substitute this with a module dependency once the module is migrated to
+    // Terraform 0.12
+    user_labels = "${merge(
+      map("tf_dependency", var.peering_completed),
+      var.user_labels)}"
```

Comment:
I don't want to add this labels by default. If `peering_completed` is not specified, then `user_labels` should be empty map.

Reply:
That's understandable. I copied-and-pasted this technique [from the `safer_mysql` submodule](https://github.com/GoogleCloudPlatform/terraform-google-sql-db/blob/master/modules/safer_mysql/main.tf#L36-L41) in this repo, so I think it suffers from the same undesirable behavior.

I am still somewhat new to Terraform, but I'm not sure I can accomplish what you want. I made two attempts at conditionally setting the label ([here](https://github.com/g5search/terraform-google-sql-db/commit/c70175fd880eb90215441d1d40722e6ec5c1d747) and [here](https://github.com/g5search/terraform-google-sql-db/commit/42cdab4bb82f2966e6df0e993a640a99fdf26333)), and both error out with:

```
conditional operator cannot be used with map values
```

Is there an example anywhere else that accomplishes this, so I can use that as a guide? I completely understand why you don't want a `tf_dependency: false` label on CloudSQL instances that aren't using this feature.

Reply:
How about this?

```hcl
locals {
  peering_completed_enabled = "${var.peering_completed != "" ? true : false}"
  user_labels_including_tf_dependency = {
    enabled = "${merge(map("tf_dependency", var.peering_completed), var.user_labels)}"
    disabled = "${var.user_labels}"
  }
}

resource "google_sql_database_instance" "default" {
  settings {
    user_labels = "${local.user_labels_including_tf_dependency["${local.peering_completed_enabled ? "enabled" : "disabled"}"]}"
  }
}
```

I'm not sure if this works correctly.
If you accept my proposal, please verify two points.

- Does this change work as expected?
- This `tf_dependency` mechanism is to ensure that this module will be run after the VPC network is created. Does this change keep this behavior?